src/screens/MyWorkouts/MyWorkouts.tsx

```diff
@@ -25,11 +29,28 @@ const MyWorkouts: React.FC<MyWorkoutsProps> = ({ navigation }) => {
   };
 
   const updateWorkout = async (payload) => {
-    await updateEditableWorkout(payload);
+    await updateEditableWorkout(payload.id, payload);
     setEditingWorkout(undefined);
     if (refreshWorkouts) await refreshWorkouts();
   };
-
+  const deleteWorkoutBlock = async (id) => {
+    await deleteEditableWorkout(id);
+    setEditingWorkout(undefined);
+    if (refreshWorkouts) await refreshWorkouts();
+  };
+  const duplicateWorkoutBlock = async (payload: EditableWorkout) => {
```

Comment:
Duplicates EditableWorkout which is passed in renderWorkoutBlock parameter as {item} in MyWorkout.tsx